loki/transformations/temporaries/stack_allocator.py

```diff
@@ -0,0 +1,1372 @@
+          INTEGER(KIND=JWIM), INTENT(IN) :: K_LD_STACK_SIZE
+          LOGICAL, TARGET, CONTIGUOUS, INTENT(INOUT) :: LD_STACK(K_LD_STACK_SIZE)
+          INTEGER(KIND=JWIM), INTENT(INOUT) :: JD_LD_STACK_USED
+          J_P_STACK_USED = JD_P_STACK_USED
+          J_P_SELECTED_REAL_KIND_13_300_STACK_USED = JD_P_SELECTED_REAL_KIND_13_300_STACK_USED
+          J_LD_STACK_USED = JD_LD_STACK_USED
+        !$loki device-present vars( p_stack, p_selected_real_kind_13_300_stack, ld_stack )
+          JD_incr = J_P_STACK_USED
+          zzx(1:nlon, 1:klev) => P_STACK(JD_incr:JD_incr + nlon*klev)
+          J_P_STACK_USED = JD_incr + klev*nlon
+          JD_incr_SELECTED_REAL_KIND_13_300 = J_P_SELECTED_REAL_KIND_13_300_STACK_USED
+          zzy(1:nlon, 1:klev) =>  &
+          & P_SELECTED_REAL_KIND_13_300_STACK(JD_incr_SELECTED_REAL_KIND_13_300: &
+              & JD_incr_SELECTED_REAL_KIND_13_300 + nlon*klev)
+          J_P_SELECTED_REAL_KIND_13_300_STACK_USED = JD_incr_SELECTED_REAL_KIND_13_300 + klev*nlon
+          JD_incr = J_LD_STACK_USED
+          zzl(1:nlon, 1:klev) => LD_STACK(JD_incr:JD_incr + nlon*klev)
+          J_LD_STACK_USED = JD_incr + klev*nlon
+
+          zzl = .false.
+          DO jl=1,nlon
+            DO jlev=1,klev
+              zzx(jl, jlev) = pzz(jl, jlev)
+              zzy(jl, jlev) = pzz(jl, jlev)
+            END DO
+          END DO
+
+        !$loki end device-present
+        END SUBROUTINE kernel
+
+    Parameters  
+    ----------  
+    block_dim : :any:`Dimension`
+        :any:`Dimension` object to define the blocking dimension.
+    horizontal : :any:`Dimension`
+        :any:`Dimension` object to define the horizontal dimension.
+    stack_name : str, optional
+        Name of the stack (default: 'STACK')
+    local_int_var_name_pattern : str, optional    
+        Local integer variable names pattern
+        (default: 'JD_{name}')
+    int_kind : str, optional
+        Integer kind (default: 'JWIM')
+    """
+
+    def adapt_temp_declarations(self, routine, temporary_arrays):
+        # make sure relevant variables are declared in their own statement
+        single_variable_declaration(routine, variables=[var.name for var in temporary_arrays])
+        # make them 'pointer' and 'contiguous'
+        for var in temporary_arrays:
+            routine.symbol_attrs[var.name] = var.type.clone(pointer=True, contiguous=True)
+        declarations = FindNodes(VariableDeclaration).visit(routine.spec)
+        # adapt declarations
+        decl_map = {}
+        for decl in declarations:
+            if decl.symbols[0] in temporary_arrays:
+                new_dimensions = as_tuple(RangeIndex((None, None)) for _ in decl.symbols[0].dimensions)
+                new_symbol = decl.symbols[0].clone(dimensions=new_dimensions)
+                if decl.dimensions is not None:
+                    decl_map[decl] = decl.clone(dimensions=as_tuple(RangeIndex((None, None)) for _ in decl.dimensions),
```

Comment:
[optional] Shorthand:
```suggestion
                    decl_map[decl] = decl.clone(dimensions=(RangeIndex((None, None),) * len(decl.dimensions)),
```